factory/src/lib.rs

```diff
@@ -0,0 +1,555 @@
+use anchor_lang::prelude::*;
+use controller::{self, Controller, MintCtx, BurnCtx};
+use members::{self, Members, Merchant};
+use anchor_lang::solana_program::clock::{self, Clock};
+use anchor_spl::token::{self, TokenAccount, Mint, Token};
+
+declare_id!("Fg6PaFpoGXkYsidMpWTK6W2BeZ7FEfcYkg476zPFsLnS");
+
+fn is_merchant(
```

Comment:
why merchants are not the part of the same program? this check would be much cleaner if it was performed at the Account level by Anchor (i.e. with pda seeds as usual). You can still have two different admins recorded in the state (for the small and big dao). I guess it may be coming from ethereum's design where it makes more sense.